src/components/Header.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {ReactElement} from 'react';
+import {View, TextStyle} from 'react-native';
+import styles from '../styles/styles';
+import Text from './Text';
+import EnvironmentBadge from './EnvironmentBadge';
+
+type HeaderProps = {
+    /** Title of the Header */
+    title?: string | ReactElement;
+
+    /** Subtitle of the header */
+    subtitle?: string | ReactElement;
+
+    /** Should we show the environment badge (dev/stg)?  */
+    shouldShowEnvironmentBadge?: boolean;
+
+    /** Additional text styles */
+    textStyles?: TextStyle[];
```

Comment:
```suggestion
    textStyles?: StyleProp<TextStyle>;
```

Let's use [style props](https://github.com/Expensify/App/blob/main/contributingGuides/TS_CHEATSHEET.md#style-props).

Reply:
@fabioh8010 It gives a type error if I use it
![image](https://github.com/Expensify/App/assets/23176449/b6a4f395-5fdc-4132-b8dc-29ef15d53eeb)

Reply:
@VickyStash Could you test if works correctly if you remove the `...`?

Reply:
Looks like `Text` component is not using `StyleProps`, that may be the reason for this errors here.

Reply:
@fabioh8010 As for me, since `textStyles` is only a part of the whole style it can't be `StyleProp<TextStyle>`.

So we have ` style={[styles.headerText, styles.textLarge, ...textStyles]}`, and `[styles.headerText, styles.textLarge, ...textStyles] ` can be typed as `StyleProp<TextStyle>`, but since textStyles is only a part of it has `TextStyle[]` type